Merging changes from US1 Implementation (Sprint 1)

dominicteh1 opened the pull request:
### What?
- Added new feature to display post content preview for each topic, for first 100 characters, appending "..." after, aligned with the post title underneath
- Updated schema to include "content" field which was added in during implementation of new feature
Resolves #6, #2, #3, #4, #33, #5 for US1

### Why?
These changes resolve US1, which is to allow students and instructors to preview the first few sentences of a post to get a quick understanding of what question is being asked. For more information, refer to US1 under "Team Bulbasaur User Stories" board.

### How?
- Navigated NodeBB site to find functions that related to topics. (Went to OH) Located function in `Category.getCategoryTopics` in `src/categories/topics.js` file, which GETs and displays all posts
- Located where the actual post content was stored, which is in `getMainPosts` function
- Introduced a new attribute "content" and took the first 100 chrs without HTML tags and called that attribute from the frontend to display the preview for each post
- Inspect element on NodeBB site to find the HTML tags `node-theme-harmony/templates/partials/topics_list.tpl`, searched the GitHub repo for those same tags to locate where the tags were in the files
- Followed guide from Prof to make a new theme "Team Bulbasaur Theme" to store the changes properly to ensure CI/CD works for future sprints

### Testing?
- Ran npm run test to check that all test cases passed
- Introduced "content" to `TopicObject.yaml` to ensure that response and schema aligned
- Fixed logic to ensure that assertions in tests were run correctly

### Screenshots
**Successful implementation of US1 feature**
![image](https://github.com/user-attachments/assets/4a0937a4-4a13-498b-b6d2-bbe912f4da8c)

**Successful test run**
![image](https://github.com/user-attachments/assets/a06c69d6-277b-4c65-86d8-5092966313b0)

### Anything Else?
Thank you

a1inachen committed:
Visually appended post content preview to topic titles (#29)

* append each topic title with content preview

* addressed lint errors

* removed console log from P1

Michaelli8899 committed:
Updated FE and BE to display Topic content preview

src/categories/topics.js: add attribute {content} to the topic object in the Categories.getCategoryTopics() function

node_modules/nodebb-theme/harmony/templates/partials/topics_list.tpl: add <div> to include topic.content in the topics_list page

dominicteh1 committed:
Fixed failing test cases after US1 feature was implemented

Michaelli8899 committed:
Adjust topics_list.tpl display method

display changes from an extra block next to the title to placing the content preview underneath the title

Michaelli8899 committed:
Modified src/categories/topics.js for text display

test whether the content has more than 100 characters and add '...' to contents that exceed the 100 character limit

bythehills committed:
Moved all changes to new theme bulbasaur

dominicteh1 committed:
Fixed linting issues

a1inachen reviewed (approved):
Yes! All makes sense and looks good to me

bythehills reviewed (approved):
lgtm !! 👍

Michaelli8899 reviewed (approved):
Yes all the code changes make sense to me

Michaelli8899 committed:
Connect UI to Frontend Repo (#32)

make changes to connect frontend changes to the nodebb-frontend-f24-team-bulbasaur repository:

link to frontend repo: https://github.com/CMU-313/nodebb-frontend-f24-team-bulbasaur

a1inachen committed:
created UserGuide file for US1

a1inachen committed:
updated UserGuide for US1

a1inachen committed:
Removed typos in UserGuide

bythehills committed:
Added test for topics_list preview

bythehills committed:
Fix lint for test/topics.js

dominicteh1 committed:
added US2 to UserGuide and edited US2 styling

bythehills committed:
US1 test fix

bythehills committed:
Fix lint

a1inachen committed:
included test link for US1 in UserGuide

a1inachen committed:
updated US1 UserGuide description
